git/index/base.py

```diff
@@ -653,7 +653,7 @@ def _to_relative_path(self, path: PathLike) -> PathLike:
             return path
         if self.repo.bare:
             raise InvalidGitRepositoryError("require non-bare repository")
-        if not str(path).startswith(str(self.repo.working_tree_dir)):
+        if not osp.normpath(str(path)).startswith(osp.normpath(str(self.repo.working_tree_dir))):
```

Comment:
Is `self.repo.working_tree_dir` usually non-normalized? To me it seems it should be left alone here, or normalized upon repository initialization.

Reply:
You're right, it's always normalized in here:
https://github.com/gitpython-developers/GitPython/blob/cfadd9e5edb8571c2c15878c8eeca91f15a69040/git/repo/base.py#L233

I've also noticed that `working_tree_dir` is never `os.PathLike` it's always either `str` or `None` - so we can also change typing for `working_tree_dir` and `_working_tree_dir` and remove `str` conversion for it. What do you think?

Reply:
To me it doesn't look it it's never `os.PathLike`, `_working_tree_dir` is initialised by many different possible branches. `expand_path` alone returns a `PathLike` for example.
On the other hand, I am not implying I *know* the type, it's a guess at best, which is probably why I'd just leave it as is to maybe increase the chance to not break something.